examples/next/src/app/address/[account]/page.tsx

```diff
@@ -0,0 +1,72 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import dynamic from 'next/dynamic';
+import { toast } from 'sonner';
+import { isAddress } from 'viem';
+
+import { useConfigStore } from '@/lib/store/use-config';
+import { useProviderStore } from '@/lib/store/use-provider';
+import { Separator } from '@/components/ui/separator';
+
+// Import dynamically to avoid SSR issues due to persisted state (see zustand stores)
+const Header = dynamic(() => import('@/components/core/header'));
+const Interact = dynamic(() => import('@/components/core/interact'));
+const TxHistory = dynamic(() => import('@/components/core/tx-history'));
+
+/**
+ * @notice An account page, where the user can interact with the contract/EOA
+ * @dev This should be displayed after searching for an account, or clicking on a link
+ * (or obviously if directly writing the address in the URL)
+ */
+export default function AccountPage({
+  params,
+}: {
+  params: { account: string };
+}) {
+  // Remember when the account has been fetched
+  const [fetched, setFetched] = useState(false);
```

Comment:
Could this just be derived state instead of a useState and effect?
```suggestion
  const fetched = !!params.account
```

Reply:
Actually, this one is a bit confusing, but it's to prevent the `useEffect` to fire `updateAccount` anytime the client/chain changes. Which would happen when the user switches chains/resets a chain. This one would be fired only on page mount.

But yeah, clearly such a hack indicates a bad design. I'll think of a different way,-; which actually now seems as obvious as removing `updateAccount` in the mentioned cases and let this `useEffect` handle it...